### PR TITLE
Clear default exp table on delete and create default exp on run create if none exists

### PR DESCRIPTION
**Pull request by rileyjbauer**
With this change, if the delete experiment API is called on the default
experiment, then the ID will also be removed from the default_experiments
table.

Additionally, if the default experiment doesn't exist and a new run is
created without an experiment, a new default experiment will be created,
and the run will be placed within this experiment.

This PR also fixes an issue in some of the backend store files where the connection to the database was not being closed. This wasn't much of an issue in production, but it appeared that in the test environment, only one connection was allowed, so failing to close the connection would cause subsequent queries to fail, claiming that the table no longer existed.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1199)

### CODE REVIEW

File: backend/src/apiserver/storage/experiment_store.go [hunk 1]
```diff
@@ -188,6 +189,11 @@ func (s *ExperimentStore) DeleteExperiment(id string) error {
 		tx.Rollback()
 		return util.NewInternalServerError(err, "Failed to delete experiment %s from table", id)
 	}
+	err = s.defaultExperimentStore.ClearDefaultExperimentId(tx, id)
```
Comment (IronPan) [inline backend/src/apiserver/storage/experiment_store.go]: how about naming this wipeoutExperimentIdIfDefault or something equvalent. 
with this name it's a bit hard to follow without context

Comment (rileyjbauer) [reply backend/src/apiserver/storage/experiment_store.go]: Yeah, you're very right. How about `UnsetDefaultExperimentIdIfIdMatches` ?

**Comment by yebrahim** [comment]
Why allow the default experiment to be deleted? If it's created automatically on init, I'd think it's a permanent resource.
I think I don't understand the model at work here, what does the default_experiments table do? Why not just have one permanent row in the experiments table with a preset id?

**Comment by rileyjbauer** [comment]
> Why allow the default experiment to be deleted? If it's created automatically on init, I'd think it's a permanent resource.
> I think I don't understand the model at work here, what does the default_experiments table do? Why not just have one permanent row in the experiments table with a preset id?

"Permanence" can be difficult to ensure in practice, and manually specifying a UUID is generally discouraged as well. The `default_experiments` table just stores the ID of the current default experiment so its UUID can be generated normally, but we can still find it without relying on its name or anything like that
